test/unit_tests/chess_compiler_tests/README.md

```diff
@@ -0,0 +1,13 @@
+There are two ways to use precompiled functions in your code.
```

Comment:
What are the pros and cons of each approach?  I think generally we should aggressively steer people to #2, unless there's no other way to accomplish what they are trying to do.

Reply:
The second line of each point has a very brief explanation for each method.
Calling a function vs. loading an elf file for a core. Recompiling also differs in each case.

I see why method two is cleaner and may be preferred when changing code. But I don't know why we should aggressively steer away from method one. Does it harm user experience or future plans/changes? We can remove it to just have method two.

Reply:
The main reason to avoid precompiled kernels is that information is replicated in multiple places.  For instance, the code for both processors that access a particular buffer need to be kept synchronized.  We use this mechanism internally, but only as the very last stage of compilation.

Reply:
It makes sense. I tried to capture your comments in the updated readme. I think it is more clear now.